spec/tree_spec.rb

```diff
@@ -1,14 +1,140 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
-  end
+describe Tree do
+    let(:tree) { Tree.new }
+    # look into rspec style guide
+    # look up relish
+
+    it 'will be a Class' do
+        expect(described_class).to eq(Tree)
+    end
+
+    describe '#initialize' do
+        it 'will generate a living tree' do
+            expect(tree.alive).to eq(true) # ask about be_true
+        end
+    end
+
+    describe '#age!' do
+        it 'will age the tree by 1 year' do
+            age = tree.age
+
+            tree.age!
+
+            expect(tree.age).to eq(age + 1)
+        end
+
+        it 'will increase the height by 5 to 10 feet' do
+            height = tree.height
+
+            tree.age!
+
+            expect(tree.height).to be_between(height + 5, height + 10)
+        end
+
+        it 'will add 1 to 6 apples' do
+            apples_count = tree.apples.count
+
+            tree.age!
+
+            expect(tree.apples.count).to be_between(apples_count + 1, apples_count + 6)
+        end
+
+        it 'will eventually die' do
+            tree.age_max.times do
+                tree.age!
+            end
+
+            expect(tree.alive).to eq(false)
+        end
+    end
+
+    describe '#add_apples!' do
+        it 'adds between 1 and 6 apples to the tree' do
+            apples_count = tree.apples.count
+
+            tree.add_apples!
+
+            expect(tree.apples.count).to be_between(apples_count + 1, apples_count + 6)
+        end
+    end
+
+    describe '#any_apples?' do
+        it 'returns true if the tree has apples' do
+            tree.add_apples
+
+            expect(tree.any_apples?).to eq(true)
+        end
+
+        it 'returns false if tree has no apples' do
+            expect(tree.any_apples?).to eq(false)
+        end
+    end
+
+    describe '#pick_an_apple!' do
+        it 'throws a NoApplesError when there are no apples' do
+            expect{ tree.pick_an_apple! }.to raise_error(NoApplesError) # this block didn't work without braces, need to look further into this
```

Comment:
My assumption about what's going on here: without the braces, `expect` just evaluates its argument, and throws an error.  When you pass in a block, it stores the block until it understands the whole test.  The raise_error part says to run the given block, and then it rescues the given error type when that block is run.